libcxx/test/std/input.output/string.streams/istringstream/istringstream.cons/string_view.mode.alloc.pass.cpp

```diff
@@ -0,0 +1,145 @@
+
+  // String-view-like
+  static_assert(
+      !std::constructible_from<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                               const int,
+                               std::ios_base::openmode,
+                               const test_allocator<CharT>>);
+  static_assert(!test_convertible<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                                  const int,
+                                  std::ios_base::openmode,
+                                  const test_allocator<CharT>>());
+
+  static_assert(
+      !std::constructible_from<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                               const SomeObject,
+                               std::ios_base::openmode,
+                               const test_allocator<CharT>>);
+  static_assert(!test_convertible<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                                  const SomeObject,
+                                  std::ios_base::openmode,
+                                  const test_allocator<CharT>>());
+
+  // Mode
+  static_assert(
+      !std::constructible_from<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                               const std::basic_string_view<CharT>,
+                               SomeObject,
+                               const NonAllocator>);
+  static_assert(!test_convertible<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                                  const std::basic_string_view<CharT>,
+                                  SomeObject,
+                                  const NonAllocator>());
+
+  // Allocator
+  static_assert(
+      !std::constructible_from<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                               const std::basic_string_view<CharT>,
+                               std::ios_base::openmode,
+                               const NonAllocator>);
+  static_assert(!test_convertible<std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>>,
+                                  const std::basic_string_view<CharT>,
+                                  std::ios_base::openmode,
+                                  const NonAllocator>());
+}
+
+#define CS(S) MAKE_CSTRING(CharT, S)
+#define ST(S) MAKE_STRING(CharT, S)
+#define SV(S) MAKE_STRING_VIEW(CharT, S)
+
+template <class CharT>
+static void test() {
+  const test_allocator<CharT> ca;
+
+  // const CharT*
+  {
+    std::basic_istringstream<CharT, std::char_traits<CharT>, test_allocator<CharT>> ss(
+        CS("zmt"), std::ios_base::binary, ca);
+    assert(ss.str() == CS("zmt"));
+    assert(ss.rdbuf()->get_allocator() == ca);
+  }
+  // std::basic_string_view<CharT>
+  {
+    const auto csv = SV("zmt");
```

Comment:
please don't use `auto` here, it does not really help to make the code more readable.